### PR TITLE
feat: folder upload

### LINKED ISSUE #69: [Feature] Folder upload
Add the ability to upload folders. 

We have this functionality in one of our apps. We take the path from the `webkitRelativePath` from the File API which can't be overridden.

This is working fix for it:

```js
Cypress.Commands.add('uploadFolder', ({ path, fileName, folderName }) => {
  cy.window().then(window => {
    class FileCopy extends File {
      get webkitRelativePath() {
        return `${folderName}/${fileName}`;
      }
    }

    window.File = FileCopy;

    cy.fixture(path).then(fileContent => {
      cy.get('input[type="file"].d-none').upload(...args);
    });
  });
});
```

### LINKED ISSUE #141: [Feature] adding webkitdirectory to support directory uploads
I would like to test a directory upload input, for that it should be enough to set the `webkitRelativePath` property in the created `File` object(s), eg:

```
File { 
name: "PANO_20190103_141758.jpg", 
lastModified: 1546521489000, 
webkitRelativePath: "images/2017/01/PANO_20190103_141758.jpg", 
size: 11591640, type: "image/jpeg" }
```

The [File](https://developer.mozilla.org/en-US/docs/Web/API/File) constructor does not support that, but you can force set it afterwards via `Object.set`. I'm using that approach already in my app if the user selects a single file instead of a directory. 

Could that be implemented?

### PR DESCRIPTION
**Pull request by abramenal**
#### Checklist:

- [ ] No linting issues
- [ ] Commits are compliant with commitizen
- [ ] CI tests have passed
- [ ] Documentation updated

#### Summary of changes

Folder upload implementation

#### TODO

- [ ] Check if this actually solves use cases
- [ ] Expose required `task` from package
- [ ] Update API reference

#### Linked issues

Closes #141
Closes #69

### CODE REVIEW

**Comment by Donaab** [comment]
@abramenal I will try to test it on my use case ... Thank you

**Comment by anark** [comment]
Hi there,
Looking into this I believe files should also have a webkitRelativePath set.

If you are handling uploaded folders you will need to be checking the items that are dropped to see if they are files or folders with something resembling the following
```
  // dataTransfer is dropped item(s)
  toArray(dataTransfer.items)
    .forEach((item) => {
      const entry = item.webkitGetAsEntry()
      // :entry can be null when we drop the url e.g.
      if (entry) {
         if (entry.isFile) {
           // Convert from FileEntry to standard File to upload
                   entry.file(
                      (file) => {
                        file.relativePath = getRelativePath(entry)
                        files.push(file)
                    })
         }
        else if (entry.isDirectory) {
          // Handle directory.  Recursively select files or do something else
        }
      }
    })
```

If only folders get the webkitRelativePath treatment then there it seems standard file uploads would fail this check as there would be not webkitFileEntry for the File item.

File: src/helpers/getFixtureInfo.js [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
   if (typeof fixtureInput === 'string') {
+    const filePath = folderName ? `${folderName}/${fixtureInput}` : fixtureInput;
+
     return {
-      filePath: fixtureInput,
+      filePath,
       encoding: '',
       mimeType: '',
       fileName: path.basename(fixtureInput),
+      ...(folderName ? { webkitRelativePath: filePath } : {}),
```
Comment (anark) [inline src/helpers/getFixtureInfo.js:12]: Should this be set on standard files as well as on folders?

**Comment by vishnuprabhu-95** [comment]
Please help us to upload folder. We have a critical functionality which uses this functionality.

**Comment by a-kriya** [comment]
@abramenal Are you open to using https://uppy.io/examples/dashboard/ for the demo/test cases here? It's the most complete and free upload solution out there and provides one widget for single or multiple files and folders. I think it can help test this new command thoroughly.

**Comment by StefanBabukov** [comment]
Are you planning to merge this soon, we would like to use this functionality?

**Comment by abramenal** [comment]
@StefanBabukov I don't think I'm able to finalize this stuff in the upcoming couple of weeks.
If anybody feels empowered to do so – I'm happy to support with review, release, etc.

**Comment by kschmarr** [comment]
@abramenal @StefanBabukov This looks like it almost made it over the finish line. Is there a chance of reviving a way to selectFolder? <3

**Comment by abramenal** [comment]
At this point I don't even know if more people have appetite for this. But if anyone here wants to finalize the implementation, I'm happy to help with code review & release process.

**Pull request closed without merging**